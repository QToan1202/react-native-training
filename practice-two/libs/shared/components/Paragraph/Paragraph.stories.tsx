import { ComponentMeta, ComponentStory } from '@storybook/react-native'

import { Paragraph } from '../index'

export default {
  title: 'components/Paragraph',
  component: Paragraph,
} as ComponentMeta<typeof Paragraph>

const Template: ComponentStory<typeof Paragraph> = (args) => <Paragraph {...args} />

export const Default = Template.bind({})
Default.args = {
  content: 'My text',
}

export const TextWithSize = Template.bind({})
TextWithSize.args = {
  content: 'My bigger text',
  size: '$2',
}

export const TextWithCustomStyle = Template.bind({})
TextWithCustomStyle.args = {
  content: 'My text is colorful',
  color: 'aqua',
}
