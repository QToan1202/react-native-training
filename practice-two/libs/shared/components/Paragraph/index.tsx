import { memo } from 'react'
import isEqual from 'react-fast-compare'

import StyledParagraph, { StyledParagraphProps } from './styles'

export type ParagraphProps = StyledParagraphProps & {
  content: string
}

const Paragraph = StyledParagraph.styleable<ParagraphProps>(({ content, ...rest }, ref) => {
  return (
    <StyledParagraph ref={ref} {...rest}>
      {content}
    </StyledParagraph>
  )
})

export default memo(Paragraph, isEqual)
