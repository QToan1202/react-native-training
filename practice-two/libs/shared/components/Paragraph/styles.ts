import { GetProps, Text, styled } from 'tamagui'

const StyledParagraph = styled(Text, {
  fontWeight: '$true',
  fontFamily: '$body',
  color: '$color.white',
})

export type StyledParagraphProps = GetProps<typeof StyledParagraph>

export default StyledParagraph
