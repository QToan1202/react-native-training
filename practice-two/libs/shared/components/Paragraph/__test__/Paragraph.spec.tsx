import { render } from '@testing-library/react-native'
import Paragraph from '../index'

test('Render text component', () => {
  const component = render(<Paragraph content="my normal text" />).toJSON()

  expect(component).toMatchSnapshot()
})

test('Text component when rendered have right content as prop', () => {
  const { getByText } = render(<Paragraph content="my normal text" />)

  expect(getByText('my normal text')).toBeTruthy()
})
