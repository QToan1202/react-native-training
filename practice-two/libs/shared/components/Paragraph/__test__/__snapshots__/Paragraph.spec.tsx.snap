// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render text component 1`] = `
<Text
  style={
    [
      {
        "color": "#fff",
        "fontFamily": "Montserrat_400Regular",
      },
      {
        "fontSize": 14,
      },
      undefined,
    ]
  }
>
  my normal text
</Text>
`;
